### ISSUE TITLE
Add a setting for multithreading

### ISSUE THREAD

**Issue by cargaji** [cms-dev/cms#1259]
**Is your feature request related to a problem? Please describe.**
Some programs, such as Python with the numpy library, make use of multiple threads by default. `isolate` can be configured to allow or disallow the creation of threads.

The current behaviour is that languages that "require multithreading", such as Java and C# (Mono) are allowed an unlimited number of threads, and all others are limited to a single thread. Additionally, whenever one of the multithreaded languages are enabled in a contest, multithreading is enabled for all languages in that contest.

**Describe the solution you'd like**
A setting to indicate the level of multithreading allowed. This could be as simple as a global value for all contests (in `cms.conf`, which in turn is passed to `isolate`), or as complex as a per-task or per-contest (even per-contest and language) setting.

**Describe alternatives you've considered**
- In the case of `numpy`, it can be [configured to use a single thread](https://stackoverflow.com/a/53224849).
- Alternatively, CMS can be edited (specifically [language.py#requires_multithreading](/cms/grading/language.py#L93)) and recompiled, but this would require a redeployment of the system.
- Either Java (JDK) or C# (Mono) can be enabled in a contest (even if the contestants should or won't use it), then multithreading is enabled for all submissions, even those that do not require multithreading.

**Comment by gollux** [comment]
I don't think that there is any reason for not enabling multithreading everywhere.

**Comment by snowynguyen** [comment]
Technically it is significantly harder for C++ competitive programmer to implement multithreading, and multithreading is in general not supported by official olympiad competitions. Therefore, I do support this issue!

**Comment by gollux** [comment]
Do you see any advantage that using multithreading would give to contestants? Remember that time limit in CMS applies to the sum of running times of all processes and threads spawned by the solution.

Re competitions: For example, current IOI rules allow multithreading.
